src/components/tutorial-meta/components/badges/components/badge/index.tsx

```diff
@@ -0,0 +1,69 @@
+import classNames from 'classnames'
+import { ProductOption } from 'lib/learn-client/types'
+import ProductIcon from 'components/product-icon'
+import { BadgeOptions } from '../..'
+import s from './badge.module.css'
+
+/**
+ * Badge wrapper provides basic badge styles and markup
+ */
+interface BadgeWrapperProps {
+  children: React.ReactNode
+  className?: string
+}
+
+function BadgeWrapper({ children, className }: BadgeWrapperProps) {
+  return <li className={classNames(s.badgeItem, className)}>{children}</li>
+}
+
+/**
+ * `getBadgeComponent` takes a map of display options
+ * and returns a default badge component that renders a label
+ * and optional icon depending on the option type
+ *
+ * This component also accepts children so it can render a composition that doesn't
+ * abide directly by the default badge display options
+ *
+ * _Note_: eventually these badges may also serve as links to filtered
+ * library views
+ */
+
+export interface BadgeComponentProps {
+  type: keyof BadgeOptions
+  className?: string
+}
+
+export function getBadgeComponent(displayOptions) {
+  function DefaultBadgeComponent({
+    type,
+    className,
+  }: BadgeComponentProps): React.ReactElement {
+    const badge = displayOptions[type]
```

Comment:
This default badge component uses the mapped data outputted by the `generateBadges` function in the next file.